documentation/src/docs/wire-protocol.md

```diff
@@ -9,104 +9,102 @@ You may obtain a copy of the License at
+-  **Begin/Middle/End**: Enum (1 byte)
+-  **Data**: A partial message is the one that was broken up when the message was sent over the wire. It is essential that for any reason, the whole message is reconstructed by reading the partial message in sequence and reassembled as whole message. It is not valid to attempt to start a new partial message before completing the previous one.
 
-### KeepAlive - Request/Reply
+## KeepAlive - Request/Reply
 
-1.  Data - uninterpreted data of the length of the message. (Usually 0 bytes)
+- **Data**: Uninterpreted data of the length of the message. (Usually 0 byte)
 
-Reading
--------
+# Reading
 
-### Read Segment - Request
 
-1.  Segment to read - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  Offset to read from - Long (8 bytes)
-3.  Suggested Length of Reply - int (4 bytes)
-    1.  This is how much data the client wants. They won't necessarily get that much.
+## Read Segment - Request
 
-### Segment Read - Reply
+1. Segment to read: A length of 2 byte String length, followed by that many bytes of Java's Modified UTF-8.
+2. Offset to read from:  8 byte Long.
+3. Suggested Length of Reply: 4 byte Integer. The clients can request for the required length to server (but the server may allot a different number of bytes).
 
-1.  Segment that was read - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  Offset that was read from - Long (8 bytes)
-3.  Is at Tail - Boolean (1 bit)
-4.  Is at EndOfSegment - (1 bit)
-5.  Data - Binary (remaining length in message)
+## Segment Read - Reply
 
-The client requests to read from a particular stream at a particular offset, it then receives one or more replies in the form of SegmentRead messages. These contain the data they requested (assuming it exists). The server may decided to give the client more or less data than it asked for, in as many replies as it sees fit.
+1.  Segment that was read: A length of 2 byte String, followed by that many bytes of Java's Modified UTF-8).
+2.  Offset that was read from: 8 byte Long.
+3.  Is at Tail: 1 bit Boolean.
+4.  Is at `EndOfSegment`: 1 bit.
+5.  Data: Binary (remaining length in the message).
 
-Appending
----------
+The client requests to read from a particular Stream at a particular Offset. It then receives one or more replies in the form of `SegmentRead` messages. These contain the data they requested (assuming it exists). The server decides on more or less data to give the client than it asked for, in as many replies as it sees fit.
 
-### Setup Append - Request
+# Appending
 
-1.  ConnectionId - UUID (16 bytes) Identifies this appender.
-2.  Segment to append to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
+## Setup Append - Request
 
-### Append Setup - Reply
+1.  `ConnectionId`: UUID (16 bytes) Identifies this appender.
+2.  Segment to append: A length of 2 byte String, followed by that many bytes of Java's Modified UTF-8.
 
```

Comment:
Before these there is also 0. RequestID which is a long used to identify this message in the reply.
and after there is 3. The delegationToken (see my earlier comment)

Reply:
Accepted suggestion